host/frontend/host-orchestrator/orchestrator/instancemanager.go

```diff
@@ -168,6 +172,20 @@ func (s *StageStartCVDServer) Run() error {
 	return err
 }
 
+type StageCreateDirIfNotExist struct {
+	Dir string
+}
+
+func (s *StageCreateDirIfNotExist) Run() error {
+	if err := os.Mkdir(s.Dir, 0755); os.IsNotExist(err) {
+		return err
+	} else if err == nil {
+		// Mkdir set the permission bits (before umask)
+		return os.Chmod(s.Dir, 0755)
+	}
+	return nil
```

Comment:
This returns a `nil` `error` for errors other than `ErrNotExist`. If `os.Mkdir` returns `ErrPermission` for permission denied, it won't enter the first or second conditional.

Was the first conditional intended to be `!os.IsNotExist(err)`? If the directory does already exist, you may want to attempt to `chown` / `chgrp` / `chmod` the directory to make sure the directory satisfies any requirements.

Reply:
for errors other than `ErrNotExist` the first condition `os.IsNotExist(err)` will test true reaching the early exit returning the error itself.

the attempt to `chmod` was meant to be done once when the directory was created the first time. But I see your point, if the directory was already there we are not going to set the relevant permission, so modified the code to set the relevant permission even if the code was there.

Reply:
> for errors other than `ErrNotExist` the first condition `os.IsNotExist(err)` will test true reaching the early exit returning the error itself.

My reading of [the `os.IsNotExist(err)` documentation](for errors other than ErrNotExist the first condition os.IsNotExist(err) will test true reaching the early exit returning the error itself.) is that the opposite is true:

> IsNotExist returns a boolean indicating whether the error is known to report that a file or directory does not exist. It is satisfied by ErrNotExist as well as some syscall errors.

Reply:
I apologize as I was wrong, got confused with the name. Your reading was the correct one.

Changed the condition to `err != nil && !os.IsExist(err)` and added a unit test showcasing the behavior where Mkdir fails with an error other than os.ErrExist.

Thanks for the review!